api/banners/serializers.py

```diff
@@ -0,0 +1,49 @@
+from api.base.utils import absolute_reverse
+from rest_framework import serializers as ser
+
+from api.base.serializers import JSONAPISerializer, DateByVersion, LinksField
+
+
+class BannerSerializer(JSONAPISerializer):
+
+    start_date = DateByVersion(read_only=True)
+    end_date = DateByVersion(read_only=True)
+    color = ser.CharField(read_only=True)
+    license = ser.CharField(read_only=True)
+    default_text = ser.SerializerMethodField()
+    mobile_text = ser.SerializerMethodField()
+
+    links = LinksField({
+        'self': 'get_absolute_url',
+        'default_photo': 'get_default_photo_url',
+        'mobile_photo': 'get_mobile_photo_url',
+    })
+
+    def get_default_photo_url(self, banner):
+        if banner.default_photo:
+            return banner.default_photo.url
+
+    def get_mobile_photo_url(self, banner):
+        if banner.mobile_photo:
+            return banner.mobile_photo.url
+
+    def get_default_text(self, banner):
+        return self.add_license(banner, banner.default_text)
+
+    def get_mobile_text(self, banner):
+        if banner.mobile_text:
+            return self.add_license(banner, banner.mobile_text)
+        return self.get_default_text(banner)
+
+    def add_license(self, banner, text):
+        if banner.license and not banner.license.lower() == 'none':
```

Comment:
Having the string 'none' be a special case seems tricky - can you just make the license field on the form not required instead of having the user typing 'none' result in no license text? If this was specifically requested all good!

Reply:
You can leave it empty as it was not required, just as I was making banners I found myself accidentally typing 'none' in the license box. I added this as a nicety in case someone else accidentally did that, but it can totally be removed.

Reply:
Naw makes sense! Woo!